src/control/cmd/daos/pool.go

```diff
@@ -364,35 +374,71 @@ func queryPool(poolHdl C.daos_handle_t, queryMask daos.PoolQueryMask) (*daos.Poo
 	}
 
 	// Preprocess queryMask, select one option for the first query
-	var firstOption string
-	if originalMask.HasOption(daos.PoolQueryOptionEnabledEngines) {
-		firstOption = daos.PoolQueryOptionEnabledEngines
-	} else if originalMask.HasOption(daos.PoolQueryOptionDisabledEngines) {
-		firstOption = daos.PoolQueryOptionDisabledEngines
-	} else if originalMask.HasOption(daos.PoolQueryOptionDeadEngines) {
-		firstOption = daos.PoolQueryOptionDeadEngines
+	queryOptions := []string{
+		daos.PoolQueryOptionEnabledEngines,
+		daos.PoolQueryOptionDisabledEngines,
+		daos.PoolQueryOptionDeadEngines,
+	}
+	var firstOption string = ""
+	for _, opt := range queryOptions {
+		if queryMask.HasOption(opt) && firstOption == "" {
+			firstOption = opt
+			continue
+		}
+		queryMask.ClearOptions(opt)
 	}
 
 	// Perform the first query to get basic information
-	if err := queryAndUpdate(firstOption); err != nil {
+	poolInfo, err := queryPoolRankLists(mockDAOS, poolHdl, queryMask)
+	if err != nil {
 		return nil, err
 	}
 
 	// Check the original query mask and update fields as needed
-	queryOptions := []string{
-		daos.PoolQueryOptionEnabledEngines,
-		daos.PoolQueryOptionDisabledEngines,
-		daos.PoolQueryOptionDeadEngines,
-	}
-
-	// Process each option sequentially
 	for _, opt := range queryOptions {
 		if originalMask.HasOption(opt) && opt != firstOption {
 			if err := queryAndUpdate(opt); err != nil {
 				return nil, err
 			}
 		}
 	}
+	poolInfo.QueryMask = originalMask
+
+	return poolInfo, nil
+}
+
+func MockQueryPool(mockDAOS *MockDAOSInterface, queryMask daos.PoolQueryMask) (*daos.PoolInfo, error) {
```

Comment:
Mocks should definitely not be in production code files. In general for Go code, we restrict mocks to their own files, or put them in test files.